cmd/erasure-server-pool.go

```diff
@@ -1980,6 +1980,34 @@ func (z *erasureServerPools) DeleteBucket(ctx context.Context, bucket string, op
 		defer lk.Unlock(lkctx)
 	}
 
+	if !opts.Force {
+		results := make(chan itemOrErr[ObjectInfo])
+
+		ctx, cancel := context.WithTimeout(ctx, time.Minute)
+		defer cancel()
+		err := z.Walk(ctx, bucket, "", results, WalkOptions{Limit: 1})
+		if err != nil {
+			s3LogIf(ctx, fmt.Errorf("unable to verify if the bucket %s is empty: %w", bucket, err))
+			return toObjectErr(err, bucket)
+		}
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case r, found := <-results:
+			if found {
+				if r.Err != nil {
+					s3LogIf(ctx, fmt.Errorf("unable to verify if the bucket %s is empty: %w", bucket, r.Err))
+					return toObjectErr(r.Err, bucket)
+				}
+				return toObjectErr(errVolumeNotEmpty, bucket)
+			}
+		}
```

Comment:
this is a bit unsafe logic, instead of relying on channel `ok` we should rely on valid `r.Err == nil` 

Do it like this

```
case r := <-results:
    if r.Err == nil { 
          return toObjectErr(errVolumeNotEmpty, bucket)
    }
    s3LogIf(ctx, fmt.Errorf("unable to verify if the bucket %s is empty: %w", bucket, r.Err))
    return toObjectErr(r.Err, bucket)
```

Reply:
If results are closed with no results, r will be the zero value with `r.Err = nil`.

Reply:
yes agree.. (by the way @harshavardhana you may have reviewed an older buggy code).